airflow/providers/docker/operators/docker.py

```diff
@@ -328,7 +331,7 @@ def _run_image_with_mounts(self, target_mounts, add_tmp_variable: bool) -> Optio
             if self.retrieve_output:
                 ret = return_value
             elif self.do_xcom_push:
-                ret = self._get_return_value_from_logs(res_lines, line)
+                ret = self._get_return_value_from_logs(lines, lines[-1])
             return ret
```

Comment:
@uranusjr 

so you were thinking a doing flavor of this, 
with the previous suggested edit (removing lines from earlier in the code) ?

it kind of looks bad, because it's too long, but it should work,
Feel free to give another suggestion to make it prettier, I'm missing the way to prettify it, if that what you meant by removing _get_return_value_from_logs

```suggestion
                return self.cli.logs(container=self.container['Id'], stdout=True, stderr=True, stream=True) if self.xcom_all else self.cli.logs(container=self.container['Id'], stdout=True, stderr=True, stream=True, tail=1)
```

Reply:
Yeah, something like this. We also need to deal with the `AirflowException` message above.

Reply:
I wonder if `"".join(logstream).splitlines()` would achieve the same as the tail-less call.

Reply:
Maybe, I tested out adding random \n into one of the bash 'echo' (without -e) commands in my example in the issue discussion,
I tried using logstream.split('\n') but it had undesired results, as some of the random `\n`'s were converted to newlines in the output, but I forgot to test splitlines() I guess

Reply:
I'll test these suggestions shortly

Reply:
I guess this small test help understand what I meant

```python
#!/usr/bin/python3

str = r"""this is
an 'herestring'
with some lines,
but now\n there's even more \n lines, but are they real?\n
string example....\nwow!!!"""
print (str.splitlines())
```

In this case, the difference between an expected results and an unexpected results lays in the `r`,
without the r""", splitlines would also process the "inline `\n`'s"

I'll test how logstream would handle splitlines